pyvista/plotting/themes.py

```diff
@@ -2742,12 +2712,12 @@ def axes(self, config: _AxesConfig):
         self._axes = config
 
     @property
-    def before_close_callback(self) -> Callable[[Plotter], None]:
+    def before_close_callback(self) -> Callable:
         """Return the default before_close_callback function for Plotter."""
         return self._before_close_callback
 
     @before_close_callback.setter
-    def before_close_callback(self, value: Callable[[Plotter], None]):
+    def before_close_callback(self, value: Callable):
```

Comment:
The return type information is actually pretty important (although I'm not sure how important it is on the theme...). I assume you removed the type signature because you removed the `plotting` dependencies. So would it work to restore these, and put the `Plotter` import inside `if typing.TYPE_CHECKING`? Do people use IDEs or other type checkers on headless installs?

It's also a bit weird if themes are `plotting.themes` and yet with carefully removed plotting dependencies. Just a thought. I also understand that functionally they belong to plotting. But doesn't importing this module already imply importing the GL stuff? I think if you `import pyvista.plotting.themes` you run `pyvista.plotting.__init__` which imports all the renderer stuff and whatnot (as it should). So perhaps we don't have to scrub this module?

Reply:
> The return type information is actually pretty important (although I'm not sure how important it is on the theme...)

I'd say it is as important as any other typing.  If I want to set something on the theme, it is useful to know what I need to set.  It makes type checking more useful if everything is as precise as possible in my experience.  I have issues with type checking in downstream applications due to lax typing in pyvista, mostly in filters.

> Do people use IDEs or other type checkers on headless installs?

I don't, but there could be a vim/emacs user who has linked in mypy or other typing functionality into their workflow on a headless machine.   The Venn diagram starts to shrink quite a bit, but it isn't hard to support this the right way, if needed (pending discussion on why this was removed).

Reply:
I meant that callbacks are trivially `Callable`. But it's not at all irrelevant what signature that callback must have. Contrast this with e.g. a filter that smells like something that takes another mesh as first argument (after `self`). And you can't just look at the docstring because---well, I guess you could look at the docstring if it were complete, with `Returns`. So yeah, I guess it's similar to other typing issues.

Reply:
Indeed I removed these annotations to remove the plotting dependency and a circular import. I'll see if I can just add this back in now that `theme.py` is under `pyvista/plotting/`

Reply:
See 2fe00b4e3328520d94e1bcdda607e58e6a5063e3